app/models/spree/line_item_decorator.rb

```diff
@@ -25,6 +25,11 @@
     end
   }
 
+  # Find line items that are from order sorted by variant name and unit value
+  scope :sorted_by_name_and_unit_value, joins(:variant => :product).
+    reorder('spree_products.name asc, spree_variants.unit_value asc').
+    select('spree_line_items.*')
+
```

Comment:
Could you please share the generated SQL query? Should be as easy as running the following in the Rails console:
```
@order.line_items.sorted_by_name_and_unit_value.to_sql
```